hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/ReplicationSupervisor.java

```diff
@@ -240,6 +240,22 @@ public void addTask(AbstractReplicationTask task) {
       return;
     }
 
+    initCounters(task);
+
+    if (inFlight.add(task)) {
+      if (task.getPriority() != ReplicationCommandPriority.LOW) {
+        // Low priority tasks are not included in the replication queue sizes
+        // returned to SCM in the heartbeat, so we only update the count for
+        // priorities other than low.
+        taskCounter.computeIfAbsent(task.getClass(),
+            k -> new AtomicInteger()).incrementAndGet();
+      }
+      queuedCounter.get(task.getMetricName()).incrementAndGet();
+      executor.execute(new TaskRunner(task));
+    }
```

Comment:
Please extract this to a new method, e.g. `addToQueue`, placed right after `initCounters` in the file.  Benefits:

- more uniform abstraction level for steps in `addTask`: `initCounters` is higher level than this block
- reduce the change triggered by extracting `initCounters`, whose code was previously above this block

Optionally the check for max. queue size can also be extracted.  Then `addTask` would look like:

```java
if (queueHasRoomFor(task)) {
  initCounters(task);
  addToQueue(task);
}
```

(Please feel free to tweak these names.)